blocks/library/embed/index.js

```diff
@@ -216,14 +229,20 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 		},
 
 		save( { attributes } ) {
-			const { url, caption, align } = attributes;
+			const { url, caption, align, type, providerNameSlug } = attributes;
 
 			if ( ! url ) {
 				return;
 			}
 
+			const embedClassName = classnames( 'wp-block-embed', {
+				[ `align${ align }` ]: align,
+				[ `type-${ type }` ]: type,
+				[ `provider-${ providerNameSlug }` ]: providerNameSlug,
```

Comment:
"align" is generally legacy for us, but for the others, let's try to use the `is-`  and `has-` convention, which limits potential clashes. (The encouragement being that people would style as `.wp-block-embed.is-provider-youtube`.

Reply:
Good proposal. Will implement that with next commit.